### PR TITLE
changes made

### PR DESCRIPTION
**Pull request by Pratyush-Saxena**
Hope these would be the changes you want...in  #99

### CODE REVIEW

**Comment by Pratyush-Saxena** [comment]
@coolaj86 pls review my pr

File: ssh-utils/ssh-pubkey.ps1 [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
 if (!(Test-Path -Path "$Env:USERPROFILE/.ssh"))
 {
-    New-Item -Path "$Env:USERPROFILE/.ssh" -ItemType Directory
+    New-Item -Path "$Env:USERPROFILE/.ssh" -ItemType Directory -Force
     #& icacls "$Env:USERPROFILE/.ssh" /inheritance:r
     #& icacls "$Env:USERPROFILE/.ssh" /grant:r "$Env:USERNAME":"(F)"
 }
 
 if (!(Test-Path -Path "$Env:USERPROFILE/.ssh/config"))
 {
```
Comment (coolaj86) [inline ssh-utils/ssh-pubkey.ps1:14]: This one is not a directory. We don't want `-Force` here.

**Review by coolaj86 — CHANGES_REQUESTED** [review]

**Comment by coolaj86** [comment]
Looks good, just one change needed in a place where the `New-Item` is being used to create a config file that _should_ fail to rewrite if it exists.

**Comment by coolaj86** [comment]
Thanks!

**Comment by Pratyush-Saxena** [comment]
@coolaj86 i have made changes as u want them to be....pls have a look

File: ssh-utils/ssh-pubkey.ps1 (hunk 1 above)
Comment (Pratyush-Saxena) [reply ssh-utils/ssh-pubkey.ps1:14]: so you dont want "-Force" anywhere in this file

**Review by Pratyush-Saxena — COMMENTED** [review]
done

File: ssh-utils/ssh-pubkey.ps1 [hunk 2]
```diff
@@ -5,14 +5,14 @@
 
 if (!(Test-Path -Path "$Env:USERPROFILE/.ssh"))
 {
```
Comment (coolaj86) [inline ssh-utils/ssh-pubkey.ps1:7]: Sorry, this line SHOULD get the -Force. It IS a directory.

```
New-Item -Path "$Env:USERPROFILE/.ssh" -ItemType Directory -Force
```

Github attached my comment to the wrong line.

**Review by coolaj86 — CHANGES_REQUESTED** [review]

**Comment by coolaj86** [comment]
Also, do you know about rebasing?

If you `git rebase -i 3` you can squash the extra commits away and have just one clean commit.

You'll then need to `git push --force origin master` on your branch.

File: ssh-utils/ssh-pubkey.ps1 (hunk 2 above)
Comment (Pratyush-Saxena) [reply ssh-utils/ssh-pubkey.ps1:7]: @coolaj86... hope its all done now

**Review by Pratyush-Saxena — COMMENTED** [review]
all done!

**Comment by Pratyush-Saxena** [comment]
> Also, do you know about rebasing?
> 
> If you `git rebase -i 3` you can squash the extra commits away and have just one clean commit.
> 
> You'll then need to `git push --force origin master` on your branch.

Actually i have never done this practically yet , but if u explain me in a procedural way , then surely i'll be able to  do that

**Comment by coolaj86** [comment]
Basically you just
- run `git rebase -i <number-of-commits>` (4 in this case, only 3 of which are squashable)
- read the instructions in the comments it pulls up
- and then run `git push --force origin master` after you're done

It is important however that you don't `squash` the very first commit. If you do that, it will squash away your authorship.

**Comment by Pratyush-Saxena** [comment]
on running ===>> "git rebase -i 3" 
It is throwing an error ==>>"fatal: invalid upstream '3' "

**Comment by coolaj86** [comment]
My bad. I think it's actually `git rebase -i HEAD~3`

**Comment by Pratyush-Saxena** [comment]
@coolaj86 is it all done now?

**Comment by coolaj86** [comment]
LGTM.

For future reference I'd suggest putting a more descriptive commit message, and reference to the issue # that's being worked on.

In this case something like "add -Force to New-Item commands #99" would have worked well.

No need to change it though, this is good to merge.

**Review by coolaj86 — APPROVED** [review]

**Comment by Pratyush-Saxena** [comment]
> LGTM.
> 
> For future reference I'd suggest putting a more descriptive commit message, and reference to the issue # that's being worked on.
> 
> In this case something like "add -Force to New-Item commands #99" would have worked well.
> 
> No need to change it though, this is good to merge.

Sure @coolaj86 ....Learned lot from you today :) ...Thanku

**Pull request merged**